workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -24,8 +24,80 @@ namespace <#= qualifiedNamespace #>
         {
             public uint ComponentId => <#= componentDetails.ComponentId #>;
 
-            public BlittableBool DirtyBit { get; set; }
-<# foreach(var fieldDetails in fieldDetailsList) { #>
+            // Bit masks for tracking which component properties were changed locally and need to be synced.
+            // Each byte tracks 8 component properties.
+<# for (var i = 0; i < (fieldDetailsList.Count / 8) + 1; i++) { #>
+            private byte dirtyBits<#= i #>;
+<# } #>
+
+            public bool IsDataDirty()
+            {
+                var isDataDirty = false;
+<# for (var i = 0; i < (fieldDetailsList.Count / 8) + 1; i++) { #>
+                isDataDirty |= (dirtyBits<#= i #> != 0x0);
+<# } #>
+                return isDataDirty;
+            }
+
+            /*
+            The propertyIndex argument counts up from 0 in the order defined in your schema component.
+            It is not the schema field number itself. For example:
+            component MyComponent
+            {
+                id = 1337;
+                bool val_a = 1;
+                bool val_b = 3;
+            }
+            In that case, val_a corresponds to propertyIndex 0 and val_b corresponds to propertyIndex 1 in this method.
+            */
+            public bool IsDataDirty(int propertyIndex)
+            {
+                if (propertyIndex < 0 || propertyIndex >= <#= fieldDetailsList.Count #>)
+                {
+                    throw new ArgumentException("propertyIndex argument out of range.");
```

Comment:
Quotes around propertyIndex

Reply:
Maybe say what the provided index was and what the acceptable range is? Could help in debugging

Reply:
Done and done. :)